jfr-streaming/src/g1GcTest/java/io/opentelemetry/contrib/jfr/streaming/G1GcMemoryMetricTest.java

```diff
@@ -41,34 +39,32 @@ class G1GcMemoryMetricTest {
                   .enableFeature(JfrFeature.GC_DURATION_METRICS)
                   .enableFeature(JfrFeature.MEMORY_POOL_METRICS));
 
-  private void usageCheck(ThrowingConsumer<MetricData> attributeCheck) {
+  @Test
+  void shouldHaveMemoryUsageMetrics() {
+    System.gc();
+    // Test to make sure there's metric data for both eden and survivor spaces.
+    // TODO: once G1 old gen usage added to jdk.G1HeapSummary (in JDK 21), test for it here too.
     jfrExtension.waitAndAssertMetrics(
         metric ->
             metric
                 .hasName(METRIC_NAME_MEMORY)
                 .hasUnit(BYTES)
                 .hasDescription(METRIC_DESCRIPTION_MEMORY)
-                .satisfies(attributeCheck),
+                .satisfies(
+                    data ->
+                        assertThat(data.getLongSumData().getPoints())
+                            .anyMatch(p -> p.getAttributes().equals(ATTR_G1_EDEN_SPACE))
+                            .anyMatch(p -> p.getAttributes().equals(ATTR_G1_SURVIVOR_SPACE))),
```

Comment:
I think it would be nice to have a method encapsulating this because it's duplicated below. It would at least read something more like 
```
.satisfies(this::verifyAttributes)
```

Reply:
+1